server/container/util/src/main/java/org/apache/james/util/concurrency/ConcurrentTestRunner.java

```diff
@@ -48,14 +54,19 @@ public ConcurrentRunnableTask(int threadNumber, BiConsumer biConsumer) {
 
         @Override
         public void run() {
+            exception = null;
             countDownLatch.countDown();
             for (int i = 0; i < operationCount; i++) {
                 try {
                     biConsumer.consume(threadNumber, i);
                 } catch (Exception e) {
                     LOGGER.error("Error caught during concurrent testing", e);
+                    exception = e;
```

Comment:
Ok so if you got 5 exception you only rethrow the last one and silently ignore all other? Why is the logic in only throwing the last one?